osbenchmark/workload/params.py

```diff
@@ -885,22 +886,48 @@ def __init__(self, workload, params, context: Context, **kwargs):
         self.field_name: str = parse_string_parameter("field", params)
         self.context = context
         self.data_set_format = parse_string_parameter("data_set_format", params)
-        self.data_set_path = parse_string_parameter("data_set_path", params)
+        self.data_set_path = parse_string_parameter("data_set_path", params, "")
+        self.data_set_corpus = parse_string_parameter("data_set_corpus", params, "")
         self.total_num_vectors: int = parse_int_parameter("num_vectors", params, -1)
         self.num_vectors = 0
         self.total = 1
         self.current = 0
         self.percent_completed = 0
         self.offset = 0
         self.data_set: DataSet = None
+        self.corpora = self.extract_corpora(self.data_set_corpus, self.data_set_format)
+
+    def get_corpora_file_path(self, name, source_format):
+        document_files = []
+        for corpus in self.corpora:
+            if corpus.name != name:
+                continue
+            filtered_corpus = corpus.filter(source_format=source_format)
+            document_files.extend([document.document_file for document in filtered_corpus.documents])
+        return document_files
 
     @property
     def infinite(self):
         return False
 
-    def _is_last_partition(self, partition_index, total_partitions):
+    @staticmethod
+    def _is_last_partition(partition_index, total_partitions):
         return partition_index == total_partitions - 1
 
+    def _validate_data_set(self):
+        if not self.data_set_path and not self.data_set_corpus:
```

Comment:
Should we check that only one is added?

Reply:
I am prioritizing data set path over corpus and only raising error if there is no way to move forward. What do you think?

Reply:
Is there any use case for this? Im thinking this could lead to confusion.

Reply:
Updated to include both condition.